utils/frame/benchmarking-cli/src/shared/mod.rs

```diff
@@ -66,6 +67,53 @@ where
 	s
 }
 
+/// A Handlebars helper to add an underscore after every 3rd character,then format it,
+/// i.e. a separator for large numbers : 123000000 -> 123_000 * WEIGHT_PER_NANOS.
+#[derive(Clone, Copy)]
+pub struct UnderscoreFormatHelper;
+
+impl handlebars::HelperDef for UnderscoreFormatHelper {
+	fn call<'reg: 'rc, 'rc>(
+		&self,
+		h: &handlebars::Helper,
+		_: &handlebars::Handlebars,
+		_: &handlebars::Context,
+		_rc: &mut handlebars::RenderContext,
+		out: &mut dyn handlebars::Output,
+	) -> handlebars::HelperResult {
+		use handlebars::JsonRender;
+		let param = h.param(0).unwrap();
+		let underscore_format_param = underscore_format(param.value().render());
+		out.write(&underscore_format_param)?;
+		Ok(())
+	}
+}
+
+/// Add an underscore after every 3rd character, then format it, i.e. a separator for large numbers:
+/// 123000000 -> 123_000 * WEIGHT_PER_NANOS.
+fn underscore_format<Number>(i: Number) -> String
+where
+	Number: std::string::ToString,
+{
+	let mut i = i.to_string().parse::<u64>().unwrap();
+	if i % constants::WEIGHT_PER_NANOS != 0 {
+		panic!("Nano seconds is the smallest unit; Weight must be divisible by it; qed");
+	}
+	i = i / constants::WEIGHT_PER_NANOS;
+
+	let mut s = String::new();
+	let i_str = i.to_string();
+	let a = i_str.chars().rev().enumerate();
+	for (idx, val) in a {
+		if idx != 0 && idx % 3 == 0 {
+			s.insert(0, '_');
+		}
+		s.insert(0, val);
+	}
+	s.push_str(" * WEIGHT_PER_NANOS");
```

Comment:
needs to be saturating or this can panic, which would be very bad

Reply:
I have checked again, I think it won't panic here, because the variable i has been parsed before , and it has divide 
 WEIGHT_PER_NANOS .  So there is no need to be saturating .

Reply:
Strong no here. I get what you are saying, the the logic above would not allow this to panic, but this is code which will live inside the runtime, and inside of Polkadot / Substrate, and we DO NOT allow panics in the runtime.

Having saturating math everywhere is NOT an accident. I will not accept this PR without saturating math everywhere.

Reply:
And I notice that there are some code like  x * WEIGHT_PER_NANOS  in the file "utils/frame/benchmarking-cli/src/storage/weights.hbs" and "utils/frame/benchmarking-cli/src/overhead/weights.hbs" ,  I'm not sure whether to update them .

Reply:
These are constants, which are evaluated at compile time. This cannot panic the runtime.